### PR TITLE
Lazy loading implementation

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
Lazy loading implemented as i proposed in [GSOC'20 Ideas list](https://github.com/Cloud-CV/GSoC-Ideas/issues/28)
List of modules for which i have included lazy loading:

- [x] Auth Module
- [x] Challenge Module
- [x] Publiclist Module

List of modules for which Refactoring is Done :

-  [x] Auth Module
-  [x] Challenge Module 
-  [x] Dashboard Module
-  [x] Home Module
-  [x] Nav Module
-  [x] Publiclist Module
-  [x] Utility Module

Work left to be done:
- [ ] Certain enhancement is required, like these modules can be further divided into sub-modules to further improve the performance of the site.

GIF showing lazy route:
ezgif com-video-to-gif (3)

### CODE REVIEW

**Comment by RishabhJain2018** [comment]
@Shekharrajak @lunayach @Sanji515 Can you please review this?

**Review by Shekharrajak — APPROVED** [review]
Good work! This should be merged before other PRs so that other PRs can be changed accordingly.

File: src/app/components/challenge/challenge.component.ts [hunk 1]
```diff
@@ -325,4 +325,25 @@ export class ChallengeComponent implements OnInit {
...
   }
 
+  inviteUsertoChallengeWrapper() {
+    const SELF = this;
+    const inviteUsertoChallenge = (e) => {
+      SELF.apiCall = (params) => {
+        const BODY = JSON.stringify(params);
+        SELF.apiService.postUrl('challenge/' + this.challenge.id + '/send-invitation/', BODY).subscribe(
+          data => {
+            SELF.globalService.showToast('success', 'User invited for the challenge successfully', 5);
```
Comment (Sanji515) [inline src/app/components/challenge/challenge.component.ts]: @Kajol-Kumari Why you added this(^) function in this PR?

Comment (Kajol-Kumari) [reply src/app/components/challenge/challenge.component.ts]: Left by mistake.will remove it.Thank you for pointing out. Though it's not in my branch code, i don't know how is it appearing here.

**Review by Sanji515 — COMMENTED** [review]
@Kajol-Kumari It seems you have refactored other modules also which are mentioned in [this](https://github.com/Cloud-CV/EvalAI-ngx/pull/299) PR. Can you also please update the PR description with these changes.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari It seems you have refactored other modules also which are mentioned in [this](https://github.com/Cloud-CV/EvalAI-ngx/pull/299) PR. Can you also please update the PR description with these changes.

@Sanji515 those modules are only refactored.Lazy loading implementation is still left to be done in those sub modules.

**Comment by Sanji515** [comment]
> > @Kajol-Kumari It seems you have refactored other modules also which are mentioned in [this](https://github.com/Cloud-CV/EvalAI-ngx/pull/299) PR. Can you also please update the PR description with these changes.
> 
> @Sanji515 those modules are only refactored.Lazy loading implementation is still left to be done in those sub modules.

Yes, can you please mention that in the PR description.

**Review by Sanji515 — APPROVED** [review]
LGTM! This can be merged :tada:

**Comment by lunayach** [comment]
@Kajol-Kumari I am seeing many changes than required for this PR. Do we need to merge any other PR before reviewing this?

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari I am seeing many changes than required for this PR. Do we need to merge any other PR before reviewing this?

@lunayach This PR includes refactoring part also as it is needed for lazy loading implementation. Also as suggested by @Sanji515 i closed the separate PR opened for refactoring and so this is the only pr containing both refactoring and lazy-loading implementation.

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**